tests/std/tests/VSO_0000000_allocator_propagation/test.cpp

```diff
@@ -785,35 +968,57 @@ void test_string_swap(const size_t id1, const size_t id2) {
 }
 
 void test_string() {
-    test_string_copy_ctor();
+    assert(pointer_counter == 0);
 
+    test_string_copy_ctor();
+    assert(pointer_counter == 0);
     test_string_copy_alloc_ctor(11, 11); // equal allocators
+    assert(pointer_counter == 0);
     test_string_copy_alloc_ctor(11, 22); // non-equal allocators
+    assert(pointer_counter == 0);
 
     test_string_copy_assign<StationaryAlloc<char32_t>>(11, 11, 11); // non-POCCA, equal allocators
+    assert(pointer_counter == 0);
     test_string_copy_assign<StationaryAlloc<char32_t>>(11, 22, 22); // non-POCCA, non-equal allocators
+    assert(pointer_counter == 0);
     test_string_copy_assign<CopyAlloc<char32_t>>(11, 11, 11); // POCCA, equal allocators
+    assert(pointer_counter == 0);
     test_string_copy_assign<CopyAlloc<char32_t>>(11, 22, 11); // POCCA, non-equal allocators
+    assert(pointer_counter == 0);
     test_string_copy_assign<CopyEqualAlloc<char32_t>>(11, 22, 11); // POCCA, always-equal allocators
+    assert(pointer_counter == 0);
 
     test_string_copy_assign_pocca_sso();
+    assert(pointer_counter == 0);
 
     test_string_move_ctor();
+    assert(pointer_counter == 0);
 
     test_string_move_alloc_ctor(11, 11); // equal allocators
+    assert(pointer_counter == 0);
     test_string_move_alloc_ctor(11, 22); // non-equal allocators
+    assert(pointer_counter == 0);
 
     test_string_move_assign<StationaryAlloc<char32_t>>(11, 11, 11); // non-POCMA, equal allocators
+    assert(pointer_counter == 0);
     test_string_move_assign<StationaryAlloc<char32_t>>(11, 22, 22); // non-POCMA, non-equal allocators
+    assert(pointer_counter == 0);
     test_string_move_assign<MoveAlloc<char32_t>>(11, 11, 11); // POCMA, equal allocators
+    assert(pointer_counter == 0);
     test_string_move_assign<MoveAlloc<char32_t>>(11, 22, 11); // POCMA, non-equal allocators
+    assert(pointer_counter == 0);
     test_string_move_assign<MoveEqualAlloc<char32_t>>(11, 22, 11); // POCMA, always-equal allocators
+    assert(pointer_counter == 0);
 
     test_string_swap<StationaryAlloc<char32_t>>(11, 11); // non-POCS, equal allocators
+    assert(pointer_counter == 0);
     // UNDEFINED BEHAVIOR, NOT TESTED - non-POCS, non-equal allocators
     test_string_swap<SwapAlloc<char32_t>>(11, 11); // POCS, equal allocators
+    assert(pointer_counter == 0);
     test_string_swap<SwapAlloc<char32_t>>(11, 22); // POCS, non-equal allocators
+    assert(pointer_counter == 0);
     test_string_swap<SwapEqualAlloc<char32_t>>(11, 22); // POCS, always-equal allocators
+    assert(pointer_counter == 0);

```

Comment:
Do I need to remove these `assert`s in `test_string`? They were used to locate the assertion failure in the main function. As the bug is solved now, I think we can just retain those in the main function?